inst/examples/app/analytics/R/read-data.R

```diff
@@ -0,0 +1,96 @@
+# define function hot to get username
+get_user <- function(session) {
+  username <- shiny::isolate(shiny::parseQueryString(session$clientData$url_search)$username)
+  if (is.null(username)) username <- "unknownUser"
+  shiny::req(username)
+  return(username)
+}
```

Comment:
I added this `get_user()` function. I just specified unknownUser when there is no session or username that can be retrieved.

Reply:
looks good!!